sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -707,7 +735,7 @@ case class AdaptiveSparkPlanExec(
    * Notify the listeners of the physical plan change.
    */
   private def onUpdatePlan(executionId: Long, newSubPlans: Seq[SparkPlan]): Unit = {
-    if (isSubquery) {
+    if (!needFinalPlanUpdate) {
```

Comment:
sorry I was long, the previous name is better. But we should update the comment here. It's not only for subquery.

Reply:
combined the comments to the definition of `shouldUpdatePlan`